components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/store/JDBCRecoveryDataStore.java

```diff
@@ -88,13 +106,19 @@ public void store(UserRecoveryData recoveryDataDO) throws IdentityRecoveryExcept
     }
 
     @Override
-    public UserRecoveryData load(User user, Enum recoveryScenario, Enum recoveryStep, String code)
-            throws IdentityRecoveryException {
+    public UserRecoveryData load(User user, Enum recoveryScenario, Enum recoveryStep, String code) throws IdentityRecoveryException {
+
+        handleRecoveryDataEventPublishing(PRE_GET_USER_RECOVERY_DATA,
+                GET_USER_RECOVERY_DATA_SCENARIO_WITH_CODE_EXPIRY_VALIDATION,null, null, code, user,
+                new UserRecoveryData(user, code, recoveryScenario, recoveryStep));
 
         PreparedStatement prepStmt = null;
         ResultSet resultSet = null;
         Connection connection = IdentityDatabaseUtil.getDBConnection(false);
         String sql;
+        UserRecoveryData userRecoveryData = null;
+        Boolean isOperationSuccess = false;
```

Comment:
Can't we use `boolean` (primitive instead of the Boolean object)?

Reply:
That would be the ideal use, but here we want to send this as an event property, which is designed to accept generic key-value pairs thus allowing only objects to be added as the value. Therefore Boolean is used instead